### PR TITLE
Misuv 2387 2

### PR DESCRIPTION
**Pull request by jack-malik**

### CODE REVIEW

**Comment by jack-malik** [comment]
There are 2 functions that need refactoring before fully migrating the page to frontend-hmrc. 1/. tableRow() and 2/. tableRow2(). They contain raw <a> tags broken into if/else logic which cannot be supported by the current implementation of the @link component. There are 3 <a> tags in total which need to be migrated after code refactoring.

File: conf/messages [hunk 1]
```diff
@@ -514,6 +514,8 @@ whatYouOwe.over-due                                                = OVERDUE
 whatYouOwe.over-due.interest.line1                                 = Interest for late payment
 whatYouOwe.over-due.interest.line2                                 = {0} to {1} at {2}%
 whatYouOwe.over-due.interest-rate                                  = Any overdue payment interest is at the {0}. The rate may have changed since the interest was first charged.
+whatYouOwe.over-due.interest-rate-1                                = Any overdue payment interest is at the {0}
+whatYouOwe.over-due.interest-rate-2                                = The rate may have changed since the interest was first charged.
```
Comment (david-walters-HMRC) [inline conf/messages:518]: more of a question but why has two brand new messages crept in, this story was to migrate not add new content?

File: app/views/WhatYouOwe.scala.html [hunk 2]
```diff
@@ -223,96 +239,119 @@ <h3 id="pre-mtd-payments-heading" class="heading-large">@messages("whatYouOwe.pr
...
+                            messageKey = "whatYouOwe.over-due.interest-rate.link", id = Some("interest-rate-link"), target = Some("_blank")).toString() + messages("whatYouOwe.over-due.interest-rate-2")),
+                        id=Some("interest-rate-para"))
+@*
+                    @p {
+                        @messages("whatYouOwe.over-due.interest-rate-1", "")
+                        @link(link = "https://www.gov.uk/government/publications/rates-and-allowances-hmrc-interest-rates-for-late-and-early-payments/rates-and-allowances-hmrc-interest-rates",
+                            messageKey = "whatYouOwe.over-due.interest-rate.link", id = Some("interest-rate-link"), target = Some("_blank"))
+                        @messages("whatYouOwe.over-due.interest-rate-2")
+                    }
+                    *@
```
Comment (david-walters-HMRC) [inline app/views/WhatYouOwe.scala.html:302]: you have removed the original code and put three different commented versions in

File: test/views/WhatYouOweViewSpec.scala [hunk 3]
```diff
@@ -58,7 +58,7 @@ class WhatYouOweViewSpec extends TestSupport with FeatureSwitching with Implicit
     val pageDocument: Document = Jsoup.parse(contentAsString(html))
 
     def verifySelfAssessmentLink(): Unit = {
-      val anchor: Element = pageDocument.getElementById("sa-note-migrated").selectFirst("a")
+      val anchor: Element = pageDocument.getElementById("payments-due-note").selectFirst("a")
```
Comment (david-walters-HMRC) [inline test/views/WhatYouOweViewSpec.scala:61]: this shouldn't be changed

File: app/views/WhatYouOwe.scala.html [hunk 4]
```diff
@@ -44,6 +44,8 @@
                 id=Some("sa-note-migrated"))
             @p(content = Html(messages("whatYouOwe.outstanding-charges-note")), id=Some("outstanding-charges-note-migrated"))
         </div>
+    } else {
+        @p(content = Html(""), id = Some("no-payments-due-note"))
```
Comment (david-walters-HMRC) [inline app/views/WhatYouOwe.scala.html:48]: don't do this, adjust the margin in the styling with something along the line of [govuk-!-padding-bottom-9]

File: app/views/WhatYouOwe.scala.html [hunk 5]
```diff
@@ -251,68 +284,70 @@ <h2 id="over-due-payments-heading" class="heading-large">@messages("whatYouOwe.o
...
                         </tbody>
                     </table>
                 </div>
-                <p>
-                    @if(!chargesList.bcdChargeTypeDefinedAndGreaterThanZero && chargesList.overduePaymentList.isEmpty && chargesList.dueInThirtyDaysList.isEmpty) {
-                        @paymentTypesDropdown
-                    }
-                </p>
+                @if(!chargesList.bcdChargeTypeDefinedAndGreaterThanZero && chargesList.overduePaymentList.isEmpty && chargesList.dueInThirtyDaysList.isEmpty) {
+                    @paymentTypesDropdown(classes = "govuk-details govuk-!-margin-top-8")
```
Comment (david-walters-HMRC) [inline app/views/WhatYouOwe.scala.html:338]: doesn't require you to pass anything in should just be @paymentTypesDropdown()

File: app/views/WhatYouOwe.scala.html [hunk 6]
```diff
@@ -251,68 +284,70 @@ <h2 id="over-due-payments-heading" class="heading-large">@messages("whatYouOwe.o
...
                 </div>
-            <div>
-                <p>
+                <div>
                     @if(!chargesList.bcdChargeTypeDefinedAndGreaterThanZero
-                    && chargesList.overduePaymentList.isEmpty) { @paymentTypesDropdown }
-                </p>
-            </div>
+                            && chargesList.overduePaymentList.isEmpty) {
+                        @paymentTypesDropdown(classes = "govuk-details govuk-!-margin-top-8")
```
Comment (david-walters-HMRC) [inline app/views/WhatYouOwe.scala.html:319]: doesn't require you to pass anything in should just be @paymentTypesDropdown()

**Pull request merged**